src/meson.build

```diff
@@ -40,7 +46,8 @@ endif
 executable(
     'dinit',
     dinit_source_files,
-    kwargs: misc_args
+    kwargs: misc_args,
+    dependencies: dinit_dependencies
```

Comment:
This could be a simple `[libselinux]` instead of a variable, because we don't many dependencies right now.

Reply:
There's no harm having a variable if it makes things clearer - putting `[libselinux]` here makes it seem like the library is always a dependency. Does this actually work correctly if libselinux isn't found? (I guess it does because the CI build passed). Anyway I guess it can stay as is it now is.

Reply:
Actually. Build output shows:
```
Run-time dependency libselinux found: NO (tried pkgconfig and cmake)
```
... why is libselinux considered a "run-time dependency"? Is it because of this? Then I think it needs to be changed back.

Reply:
libselinux is a "run-time dependency" in the sense that it is being searched for in order to use it when compiling installable executables.

It would be a 
```
Build-time dependency libselinux found: NO
```

if it was searched for using `dependency('libselinux', native: true)` and its purpose was during a cross-compile to compile native executables that required libselinux and could be used as e.g. code generator programs.

Reply:
> There's no harm having a variable if it makes things clearer - putting `[libselinux]` here makes it seem like the library is always a dependency. Does this actually work correctly if libselinux isn't found? (I guess it does because the CI build passed). Anyway I guess it can stay as is it now is.

Dependencies can be looked up only optionally, in this case:

```meson
libselinux_dep = dependency('libselinux', version : '>= 2.1.9', required : support_selinux)
```

it will only be looked up when the "support-selinux={enabled|disabled|auto}" build option permits it. The return value can be checked to see if it was indeed found -- and either way, you can use it anywhere a dependency is a valid parameter type.

- If libselinux was found, it expands to the correct compile/link arguments for libselinux. 
- If libselinux was NOT found, it expands to nothing, and is a no-op. 

So it is always safe to pass it even if it isn't used, greatly reducing the amount of control flow necessary for converting build configuration intents into compilation commands.

Reply:
@eli-schwartz 
> libselinux is a "run-time dependency" in the sense that it is being searched for in order to use it when compiling installable executables.

I think maybe I understand what you are saying, but it seems very much at odds with what I would normally think of as a run-time vs build-time dependency. To me a dependency that is used when compiling the final executable(s), or at any stage in the lead-up to that, is a build-time dependency; i.e. a build-time dependency is something that is required when the application is built, and a run-time dependency is required when the application is run.

Anyway, I guess that means this is fine as is, and that the "run-time" message is correct as far as meson is concerned - thanks for clarifying.

Reply:
It's a bit of weird wording -- I certainly didn't choose it lol. :)

I went and looked up where it was introduced in the git history -- @Ericson2314 introduced it in https://github.com/mesonbuild/meson/commit/07777e15d47dbddaf849d24b3a30c85745c533ca as part of a refactor that shouldn't actually have affected this at all and I do *not* know why! Before that commit, it would list "Dependency libselinux found: NO" except in a cross build, when it would say "Cross dependency libselinux found: NO".